### ISSUE TITLE
the way access nested scala singleton from java code changes in JDK11

### ISSUE THREAD

**Issue by docete** [scala/bug#12111]
## reproduction steps

I have the following scala code (in scala 2.11.12)
```scala
package pkg1

object outer {
    object inner {
        def foo() = ???
    }
}
```
And the following java code to call function foo
```java
package pkg2;

import pkg1.outer;

public class Test {
    public void bar() {
        outer.inner$.MODULE$.foo();
    }
}
```
for JDK8, it works. But for JDK11, the compiler complain "cannot find symbol inner$"

## problem

I checked the class files and found the decompiled codes are exactly the same.

**Comment by lrytz** [comment]
I couldn't reproduce this. Can you show step by step what you did, which versions?

```
$> cat Test.scala
package pkg1

object outer {
    object inner {
        def foo() = println("hi")
    }
}
$> cat Test.java
package pkg2;

import pkg1.outer;

public class Test {
  public static void main(String[] args) {
        outer.inner$.MODULE$.foo();
  }
}
$> scalac -2.11.12 Test.scala -d .
$> java -version
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
$> javac -cp .:$(cs fetch -p org.scala-lang:scala-library:2.11.12) -d . Test.java
$> java -cp .:$(cs fetch -p org.scala-lang:scala-library:2.11.12) pkg2.Test
hi
$>
```

**Comment by docete** [comment]
@lrytz Seems the simplified cases couldn't reproduce this. I will close this and find another case. Thank you!

**Comment by lqhuang** [comment]
@docete Hey, did you still meet similar problems. I run into a very similar one. Let me try to make a minimal reproducible case.

**Comment by som-snytt** [comment]
@lqhuang https://github.com/scala/bug/issues/9714 was fixed in 2.13.13

That is for "mixed compilation", not "separate compilation".

**Comment by som-snytt** [comment]
The fix has a test. I verified that it also works for nested object, or from Java
```
        public static p.Module.Target$ target() {
                return p.Module.Target$.MODULE$;
        }
```
I don't recall the details, so I guessed at where to write `$`.

I will close this as duplicate, but please update if there is more info.
